### ISSUE TITLE
Questions about shippable configuration for pull request

### ISSUE THREAD

**Issue by JoeRattanamaneesilp** [Shippable/support#3938]
Hi Shippable support team,

I'm working on reviewing the shippable configuration in the project. Could  you please have a look some questions that I have?

1. Could you please clarify whether there is a difference between using a double-quote and not using it in a variable in shippable.yml?
e.g. 
- `if [ "$IS_PULL_REQUEST" == false ]; then NODE_ENV=production npm run build; fi`
and
- `if [ $IS_PULL_REQUEST == false ]; then NODE_ENV=production npm run build; fi`

Regarding this page, it is showing double-quote when using a variable in a condition.
http://docs.shippable.com/ci/pull-request-builds/#customizing-config-for-pull-requests

However, we have been using it without double-quote in the projects for a while. Please confirm if we need to change. 

2. One condition in shippable.yml that I can see in the project is `if [ "$BRANCH" == "master" ] `
However, I don't see it in the table on this page http://docs.shippable.com/ci/pull-request-builds/#customizing-config-for-pull-requests 
Could you please clarify if we are using an invalid variable? 

3. Every time when we commit and push an update to a feature branch which has a pull request, shippable will run CI for two jobs.
- master 
- feature branch
![image](https://user-images.githubusercontent.com/14726381/32421040-fd6927f6-c2e7-11e7-868d-efa071196439.png)
Could you please clarify what the difference between those two jobs is? Why does Shippable have to run the job of master? Is it mandatory?

Regards,
Joe

**Comment by trriplejay** [comment]
1. This is not a requirement. the line is directly translated to a bash script line, so the only consideration is whether or not the yml is still valid and parse-able, and in this case it works both ways.
2. the ENVs on that page are a subset of the total available ENVs.  $BRANCH is definitely valid, and can be found in this more complete table: http://docs.shippable.com/ci/env-vars/
3. Job 3067 is being run against the opened pull request, while 3068 is being run against the master branch after the PR has been merged.  Ideally there wouldn't be a difference, and the results of the PR build should be a good indicator that the code could be safely merged.  However, I can imagine a scenario in which two feature branches open PRs against master at the same time. It's possible that individually they each have a successful PR build, but then once both are merged, they could possibly impact each other and result in a failed master build.  If you think that running the master branch build is unnecessary, there are several ways you can change your settings to prevent this:

- use the `branches: except` section of the yml to prevent master branch from running at all.  You can read about this here: http://docs.shippable.com/ci/specify-branches/
- go into your project settings page, and disable 'commit' builds, so that only PRs will trigger shippable builds

screenshot of project settings:
![image](https://user-images.githubusercontent.com/6869398/32457536-53242308-c2de-11e7-838f-fba42887a0f9.png)


Let me know if you have any more questions!

**Comment by JoeRattanamaneesilp** [comment]
Hi @trriplejay ,

Thanks so much for your clarification.

Regarding the question no.3, the job no.3068 was run at the same time with 3067. They were in a pull request and hadn't been merged to master yet. Once it is merged to master, it will have this kind of logo with master branch. The screenshot below is showing three different jobs that I have noticed.

![image](https://user-images.githubusercontent.com/14726381/32471537-e9ddbb5a-c3b1-11e7-84d0-f48479446c50.png)

So I think the job no.3100 is not necessary for my project as it seems to be covered in the job no.3099. I will change the setting as your suggestion. 

Cheers,
Joe

**Comment by trriplejay** [comment]
Ah, my mistake. Shippable automatically runs a build for any branch that contains the shippable.yml, and we also run builds for pull requests, so if you make a commit to a branch that also has an open pull request, github sends 2 webhooks, and shippable triggers 2 builds.

You can ignore builds like 3100 if you prefer, but one nice thing about the pull request build is that it will internally merge the changes to the head branch before running your unit tests.  This way you're not just testing the branch, you're testing master as though the branch had been merged into it.  This might give you a more informative test result than the branch alone would give.  

Ultimately its up to you, and depends on how exactly you are configuring your tests. Let us know if you have any more questions!

**Comment by JoeRattanamaneesilp** [comment]
Hi @trriplejay,

Thanks for your clarification. The advantage of having a build against master in a pull request is very good.

Please let me share the setting in my project.  I think it is another solution.

For the github setting in our project, we have a setting to block merging if a feature branch is behind the master branch. The user has to update the branch first by clicking the button. 
![image](https://user-images.githubusercontent.com/14726381/32521813-13c34734-c469-11e7-99d9-764ac2ff2213.png)

After clicking it, the master branch will be auto-merged to the feature branch.
![image](https://user-images.githubusercontent.com/14726381/32521833-27098524-c469-11e7-96c0-58012baf76ed.png) 

Then, a new runCi job will be auto-triggered again. 
![image](https://user-images.githubusercontent.com/14726381/32521877-4771dadc-c469-11e7-881c-f04cba2583f9.png)

Thanks for your help in this issue.

Regards,
Joe

**Comment by trriplejay** [comment]
that makes sense. in that case, disabling pull request builds for the project should be fine, as the branch itself will be running with the latest from master.  You can just toggle off the button in the screenshot i originally posted for the "pull requests" section.
